### DISCUSSION TITLE
Can I publish to root gh-pages (username.github.io) from repo other than github.com/username/username.github.io using GitHub Actions?

### DISCUSSION THREAD

**Discussion by benlindsay** [quarto-dev/quarto-cli#3512] [Q&A] [answered]
I have a repo, https://github.com/benlindsay/quarto-blog, that contains my blog source code. I'd like to use the blog rendered from that repo to show up as https://benlindsay.github.io, not https://benlindsay.github.io/quarto-blog, which is where the default GitHub actions setup puts it. I currently/previously have/had a Gatsby site at https://github.com/benlindsay/gatsby-site, and I manually published that with a script that looks like this:

```bash
repo="git@github.com:benlindsay/benlindsay.github.io.git"
rm -rf benlindsay.github.io
git clone "$repo"
mv benlindsay.github.io/.git public/
rm -rf benlindsay.github.io
(cd public && git add . && git commit -m "$(date)" && git push origin master)
```

Which takes the site content which was generated in the `public/` directory of my `gatsby-site` repo and updates the `benlindsay.github.io` repo with that content. Do I need to make a bash script like this for my quarto site too, or is there a less hacky way of accomplishing what I'm looking for? Can I set that up with github actions? If I want to use github actions, is the expected way to do this to set up the source code in the `benlindsay.github.io` repo rather than an arbitrarily-named repo?

**Comment by mcanouil** [comment, marked as answer]
The Quarto GitHub action is a simple and minimalist action with low privileges requirements.

If you want to build from repo A and push results to repo B, then you have to write those steps in your GitHub Action workflow.
Default secret token would allow you to do the git push step to another repository, thus you'll need to set a new token with sufficient permission.

**Reply by benlindsay** [reply]
Thanks for the info! For now I guess I'll just try setting up my repo in https://github.com/benlindsay/benlindsay.github.io for simplicity
